scantool/diag_l0_elm.c

```diff
@@ -824,6 +897,9 @@ elm_send(struct diag_l0_device *dl0d,
 	if (len <= 0)
 		return diag_iseterr(DIAG_ERR_BADLEN);
 
+	if ((dev->protocol & DIAG_L1_ISO9141) && len <= 3)
```

Comment:
Previously, elm_send() required at least 1 payload byte. Now, in ISO9141, it gets the header from L2, so it requires 3 header bytes + 1 payload byte. This change preserves the previous requirement of at least 1 payload byte.